### DISCUSSION TITLE
Gunicorn Socket Error

### DISCUSSION THREAD

**Discussion by derek8bai** [miguelgrinberg/Flask-SocketIO#1981] [General]
We have a Flask Server deployed on Elastic Beanstalk that seems to just hang randomly with this error. Does anyone know what is going on?

[2023-04-28 00:02:40 +0000] [4238] [ERROR] Socket error processing request.
Apr 28 00:02:40 ip-172-31-7-208 web: Traceback (most recent call last):
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/workers/gthread.py", line 271, in handle
Apr 28 00:02:40 ip-172-31-7-208 web: keepalive = self.handle_request(req, conn)
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/workers/gthread.py", line 343, in handle_request
Apr 28 00:02:40 ip-172-31-7-208 web: util.reraise(*sys.exc_info())
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/util.py", line 626, in reraise
Apr 28 00:02:40 ip-172-31-7-208 web: raise value
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/workers/gthread.py", line 329, in handle_request
Apr 28 00:02:40 ip-172-31-7-208 web: resp.write(item)
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/http/wsgi.py", line 326, in write
Apr 28 00:02:40 ip-172-31-7-208 web: self.send_headers()
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/http/wsgi.py", line 322, in send_headers
Apr 28 00:02:40 ip-172-31-7-208 web: util.write(self.sock, util.to_bytestring(header_str, "latin-1"))
Apr 28 00:02:40 ip-172-31-7-208 web: File "/var/app/venv/staging-LQM1lest/lib64/python3.8/site-packages/gunicorn/util.py", line 287, in write
Apr 28 00:02:40 ip-172-31-7-208 web: sock.sendall(data)
Apr 28 00:02:40 ip-172-31-7-208 web: OSError: [Errno 9] Bad file descriptor

**Comment by miguelgrinberg** [comment]
This is likely a Gunicorn bug. In many cases this type of error does not cause any real problems. It could be that the client just went away.
